### ISSUE TITLE
How can i install PyTorch3D in Windows 10

### ISSUE THREAD

**Issue by charuyskaya** [facebookresearch/pytorch3d#614]
## ❓ How can i install PyTorch3D on Windows 10?

I've already installed torch and torchvision, using command `pip install torch==1.8.0+cu111 torchvision==0.9.0+cu111 torchaudio===0.8.0 -f https://download.pytorch.org/whl/torch_stable.html
`. But i can't install pytorch3d. Could you please help me?

Here below i put my logs.
Thank you.

```
pip install "git+https://github.com/facebookresearch/pytorch3d.git@stable"
Collecting git+https://github.com/facebookresearch/pytorch3d.git@stable
  Cloning https://github.com/facebookresearch/pytorch3d.git (to revision stable) to c:\users\79857\appdata\local\temp\pip-req-build-qio4m71m
  Running command git clone -q https://github.com/facebookresearch/pytorch3d.git 'C:\Users\79857\AppData\Local\Temp\pip-req-build-qio4m71m'
  error: RPC failed; curl 56 OpenSSL SSL_read: Connection was reset, errno 10054
  error: 35 bytes of body are still expected
  fetch-pack: unexpected disconnect while reading sideband packet
  fatal: early EOF
  fatal: fetch-pack: invalid index-pack output
WARNING: Discarding git+https://github.com/facebookresearch/pytorch3d.git@stable. Command errored out with exit status 128: git clone -q https://github.com/facebookresearch/pytorch3d.git 'C:\Users\79857\AppData\Local\Temp\pip-req-build
-qio4m71m' Check the logs for full command output.
ERROR: Command errored out with exit status 128: git clone -q https://github.com/facebookresearch/pytorch3d.git 'C:\Users\79857\AppData\Local\Temp\pip-req-build-qio4m71m' Check the logs for full command output.
```

**Comment by bottler** [comment]
It may be quite complicated to get PyTorch3D working on windows, so I would first say that if you just want to briefly play with PyTorch3D you might prefer to use Google Colab.

The error message you are getting suggests a network problem between you and github. I don't know what is causing that. But even if the network was fine I think the command will have further problems.

You need to set up all the dependencies, for which I recommend conda, and then build from source. Rather than installing straight from github with pip, I think you are better off cloning this repository, checking out stable, and then proceeding with something like `pip install .` or `python setup.py install`. I think this will make it easier to understand and fix build errors. It will also mean that if you want to try multiple times you won't be relying on the network connection to work each time.

For different versions of PyTorch, users have found different changes have had to be made to the PyTorch header files to make PyTorch3D compile. No one has yet let us know what was necessary for PyTorch 1.7.0 or PyTorch 1.8.0. PyTorch 1.8.0 is the latest and I don't know if any changes will be required; it is possible things have been fixed so that they aren't. The "safe" method would be to install PyTorch 1.6.0 and to make the changes described in Issue #323. This would be done with something like `conda install -c pytorch pytorch=1.6.0 torchvision  cudatoolkit=11.1` in a new conda environment.

**Comment by charuyskaya** [comment]
@bottler hi!

I've installed Ubuntu today. And now i have another problem:
`RuntimeError: Not compiled with GPU support`

My environment is (a part of):
```
pytorch                   1.7.1           py3.8_cuda10.2.89_cudnn7.6.5_0    pytorch
pytorch3d                 0.4.0                    pypi_0    pypi
python                    3.8.8                hdb3f193_4  
torchvision               0.8.2                py38_cu102    pytorch
```
I ran this code to test CUDA:
```
python -c 'import torch; from torch.utils.cpp_extension import CUDA_HOME; print(torch.cuda.is_available(), CUDA_HOME)'

True /usr/local/cuda-10.2
```
My Nvidia driver version is 460.39 and compatible with cuda 10.2.
And I also try use: `FORCE_CUDA=1 ` and it doesn't work for me :(

Tell me pls where is my possible mistake?
Thank you!

**Comment by bottler** [comment]
This is a much easier situation. You are now on Linux, and using conda. CUDA is working but you are having trouble compiling cuda code. This means that when you build PyTorch3D it is only building for the CPU. There are two ways forward:

1) Install a prebuilt version of PyTorch3D. This should be very easy. Just start a new environment and proceed as in INSTALL.md.

2) Work out why the build is failing. What happens when you run `nvcc`? Maybe you need to add `$CUDA_HOME/bin` to the PATH. If `nvcc` exists, then can you try again with FORCE_CUDA=1 and paste the output?

**Comment by charuyskaya** [comment]
@bottler thank you for quick answers! 

I tried both variants, but still have errors :(

1. This error i got after creation a new environment with [prebuilt version pytorch3d](https://github.com/facebookresearch/pytorch3d/blob/master/INSTALL.md#3-install-wheels-for-linux).

![Error_after_installing_prebuilt_version](https://user-images.githubusercontent.com/26479036/112532303-bc25eb80-8db9-11eb-9cec-f0a4ec98c944.png) 

2. This is error from my previous message, I attached logs with information you asked.

![CurrentError](https://user-images.githubusercontent.com/26479036/112532300-baf4be80-8db9-11eb-846c-a44c078bfbb7.png)

Here you can see that i have variable `FORCE_CUDA=1`
![Check_FORCE_CUDA](https://user-images.githubusercontent.com/26479036/112532296-b9c39180-8db9-11eb-952a-e0112ff90cca.png)

Help me please to overcome this errors

**Comment by bottler** [comment]
For 1 - please give us the full output of `conda list`. Maybe tell us exactly which commands you ran. I think you have ended up with the wrong build of PyTorch3D for your PyTorch version.

For 2 I am confused. Please tell us exactly which commands you ran to build the library and what they output. Note that running `FORCE_CUDA=1` on its own is not useful - it just sets a shell variable, not an environment variable.  `set` shows you shell variables, whether or not they are exported to the environment. The significance of `FORCE_CUDA` being `1` is that if this is in the *environment* when PyTorch3D is *built* then it will cause PyTorch3D to be built even if `torch.cuda.is_available()` is `False`. For example `FORCE_CUDA=1 pip install .` on one line or `export FORCE_CUDA=1` on a line before the build command. I don't know why it would be relevant in your case.

Is there a mixture of cuda 10.2 and 10.1?

**Comment by charuyskaya** [comment]
@bottler, thank you!

I've fixed my cuda versions, now it is 10.2 everywhere.

But when i try to pip install pytorch3d from source i have new errors:
`error: command '/home/charuyskaya/anaconda3/envs/main/bin/x86_64-sarc-linux-gnu-cc' failed with exit status 1`

and 
`x86_64-sarc-linux-gnu-gcc.bin: error trying to exec 'cc1plus': execvp: No such file or directory
error: command '/home/charuyskaya/anaconda3/envs/main/bin/x86_64-sarc-linux-gnu-cc' failed with exit status 1`

or this error:
`Command '['which', '/home/charuyskaya/anaconda3/envs/main/bin/x86_64-conda_cos6-linux-gnu-c++']' returned non-zero exit status 1.`

Conda list here below:
<pre># Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
binutils_linux-cos6-x86_64 2.28                 h5f4b72c_0    tsnyder
blas                      1.0                         mkl  
ca-certificates           2021.1.19            h06a4308_1  
certifi                   2020.12.5        py37h06a4308_0  
chumpy                    0.70                     pypi_0    pypi
cudatoolkit               10.2.89              hfd86e86_1    main
cudnn                     7.6.5                cuda10.2_0    main
cycler                    0.10.0                   pypi_0    pypi
decorator                 4.4.2                    pypi_0    pypi
face-alignment            1.3.3                    pypi_0    pypi
freetype                  2.10.4               h5ab3b9f_0  
future                    0.18.2                   pypi_0    pypi
fvcore                    0.1.3.post20210223    pyhd8ed1ab_0    conda-forge
gcc_linux-cos6-x86_64     6.3.0                h6ab855c_0    tsnyder
imageio                   2.9.0                    pypi_0    pypi
intel-openmp              2020.2                      254  
iopath                    0.1.8                      py37    iopath
jpeg                      9b                   h024ee3a_2  
kiwisolver                1.3.1                    pypi_0    pypi
lcms2                     2.11                 h396b838_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libffi                    3.3                  he6710b0_2  
libgcc-ng                 9.1.0                hdf63c60_0  
libpng                    1.6.37               hbc83047_0  
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.2.0                h85742a9_0  
libwebp-base              1.2.0                h27cfd23_0  
llvmlite                  0.36.0                   pypi_0    pypi
lz4-c                     1.9.3                h2531618_0  
matplotlib                3.4.0                    pypi_0    pypi
mkl                       2020.2                      256  
mkl-service               2.3.0            py37he8ac12f_0  
mkl_fft                   1.3.0            py37h54f3939_0  
mkl_random                1.1.1            py37h0573a6f_0  
ncurses                   6.2                  he6710b0_1  
networkx                  2.5                      pypi_0    pypi
numba                     0.53.1                   pypi_0    pypi
numpy                     1.20.1                   pypi_0    pypi
numpy-base                1.19.2           py37hfa32c7d_0  
nvidiacub                 1.10.0                        0    bottler
olefile                   0.46                     py37_0  
opencv-python             4.5.1.48                 pypi_0    pypi
openssl                   1.1.1k               h27cfd23_0  
pillow                    8.1.2            py37he98fc37_0  
pip                       21.0.1           py37h06a4308_0  
portalocker               2.2.1            py37h06a4308_0  
pyparsing                 2.4.7                    pypi_0    pypi
python                    3.7.10               hdb3f193_0    main
python-dateutil           2.8.1                    pypi_0    pypi
pywavelets                1.1.1                    pypi_0    pypi
pyyaml                    5.4.1                    pypi_0    pypi
readline                  8.1                  h27cfd23_0  
scikit-image              0.18.1                   pypi_0    pypi
scipy                     1.6.2                    pypi_0    pypi
setuptools                52.0.0           py37h06a4308_0  
six                       1.15.0           py37h06a4308_0  
sqlite                    3.35.2               hdfb4753_0  
tabulate                  0.8.9            py37h06a4308_0  
termcolor                 1.1.0            py37h06a4308_1  
tifffile                  2021.3.17                pypi_0    pypi
tk                        8.6.10               hbc83047_0  
torch                     1.6.0                    pypi_0    pypi
torchvision               0.7.0                    pypi_0    pypi
tqdm                      4.59.0             pyhd3eb1b0_1  
wheel                     0.36.2             pyhd3eb1b0_0  
xz                        5.2.5                h7b6447c_0  
yacs                      0.1.6                      py_0    conda-forge
yaml                      0.2.5                h7b6447c_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.5                h9ceee32_0  
</pre>

**Comment by charuyskaya** [comment]
@bottler so i tried one more time to install prebuilt version with python3.6 and it works!

Thank you for help, i don't have any questions :)
